### PR TITLE
Support median in Groupby.aggregate

### LINKED ISSUE #592: Add median support to GroupBy.aggregate

### PR DESCRIPTION
**Pull request by hendrikmakait**
- Blocked by ~https://github.com/dask/dask/pull/10835~ ~dask/dask#10840~
- Closes #592 
- Sibling to https://github.com/dask/dask/pull/10832

### CODE REVIEW

**Comment by phofl** [comment]
the dask/dask pr is merged

**Comment by crusaderky** [comment]
Empty commits with `[skip-caching]` don't seem to be doing anything. See https://github.com/dask-contrib/dask-expr/pull/804

**Comment by hendrikmakait** [comment]
CI failure is #806

**Comment by crusaderky** [comment]
@hendrikmakait I'm confused.

These are still failing in the dask-expr CI:
```
FAILED ../../../micromamba/envs/dask-expr/lib/python3.9/site-packages/dask/dataframe/tests/test_groupby.py::test_groupby_aggregate_partial_function_unexpected_args[disk-<lambda>0] - AssertionError: Regex pattern did not match.
 Regex: "doesn't support positional arguments|'Series' object cannot be interpreted as an integer"
 Input: "cannot convert the series to <class 'int'>"
FAILED ../../../micromamba/envs/dask-expr/lib/python3.9/site-packages/dask/dataframe/tests/test_groupby.py::test_groupby_aggregate_partial_function_unexpected_args[disk-<lambda>1] - AssertionError: Regex pattern did not match.
 Regex: "doesn't support positional arguments|'Series' object cannot be interpreted as an integer"
 Input: "cannot convert the series to <class 'int'>"
FAILED ../../../micromamba/envs/dask-expr/lib/python3.9/site-packages/dask/dataframe/tests/test_groupby.py::test_groupby_aggregate_partial_function_unexpected_args[tasks-<lambda>0] - AssertionError: Regex pattern did not match.
 Regex: "doesn't support positional arguments|'Series' object cannot be interpreted as an integer"
 Input: "cannot convert the series to <class 'int'>"
FAILED ../../../micromamba/envs/dask-expr/lib/python3.9/site-packages/dask/dataframe/tests/test_groupby.py::test_groupby_aggregate_partial_function_unexpected_args[tasks-<lambda>1] - AssertionError: Regex pattern did not match.
 Regex: "doesn't support positional arguments|'Series' object cannot be interpreted as an integer"
 Input: "cannot convert the series to <class 'int'>"
```

However, if I run manually in dask/dask, I get everything green. Same pandas version...

File: dask_expr/_reductions.py [hunk 1]
```diff
@@ -447,6 +447,13 @@ def _divisions(self):
     def _chunk_cls_args(self):
         return []
 
+    @property
+    def should_shuffle(self):
+        sort = getattr(self, "sort", False)
+        return not (
+            not isinstance(self.split_out, bool) and self.split_out == 1 or sort
+        )
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_reductions.py:455]: this doesn`t cover the bool case

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

File: dask_expr/_reductions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rjzamora) [reply dask_expr/_reductions.py:455]: Sorry my suggestion was off, but I was pointing out the sort check is wrong (I think).

[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_reductions.py:455]: Why can't we use a tree reduction? The modus is to reduce to one partition and then sort, so we can not shuffle if sort=True, we have to reduce to one partition

[the remark this answers is not in the document]
Comment (rjzamora) [reply dask_expr/_reductions.py:455]: I'm sorry for leaving a "this is a bug" comment, and then getting pulled away.

I think some of my confusion comes from that fact that dask/dask **allows** the tree reduction path to be used with `split_out > 1`, but that algorithm will produce incorrect results if `sort=True`.

In dask-expr, we do **not** allow the tree reduction to be used with `split_out > 1`, so the sort check seems unnecessary/confusing. If `split_out == `, then we don't really care about `sort`.

**Pull request merged**